Source/com/drew/metadata/mp4/boxes/TimeToSampleBox.java

```diff
@@ -49,7 +49,14 @@ public TimeToSampleBox(SequentialReader reader, Box box) throws IOException
 
     public void addMetadata(Mp4VideoDirectory directory)
     {
-        float frameRate = (float) Mp4HandlerFactory.HANDLER_PARAM_TIME_SCALE/(float)entries.get(0).sampleDelta;
+        float sampleCount = 0;
+
+        for (EntryCount ec : entries) {
+            sampleCount += ec.sampleCount;
+        }
+
+        float frameRate = (float) Mp4HandlerFactory.HANDLER_PARAM_TIME_SCALE/((float) Mp4HandlerFactory.HANDLER_PARAM_DURATION / sampleCount);
```

Comment:
I assume `entries` will always have one item, otherwise we'll be dividing by zero.